deploy/docs/Best_Practices.md

```diff
@@ -0,0 +1,112 @@
+
+Users can specify their custom regex to detect the first line of a multiline log and parse the logs correctly.
+
+New parsers can be defined under the `parsers` key of the fluent-bit configuration section in `values.yaml` as follows:
+
+```
+parsers:
+  enabled: true
+  regex:
+    - name: multi_line
+    regex: (?<log>^{"log":"\d{4}-\d{1,2}-\d{1,2} \d{2}:\d{2}:\d{2}.*)
+    - name: new_parser_name
+    ## This parser matches lines that start with time of the format : 07:14:12
+    regex: (?<log>^{"log":"\d{2}:\d{2}:\d{2}.*)
+```
+
+The regex used for `Parser_Firstline` needs to have at least one named capture group.
+
+In order to use the newly define parser to detect the first line of multiline log, change the `Parser_Firstline` parameter in the `Input plugin` configuration of fluent-bit:
+
+```bash
+Parser_Firstline new_parser_name
+```
+
+Users can also use optional-extra parser to interpret and structure multiline entries.
+When Multiline is On, if a line matched `Parser_Firstline`, continuation lines will be matched against `Parser_N` parsers (just if a previous Parser_Firstline match exists).
+
+```bash
+Parser_Firstline multi_line
+Parser_1 optional_parser
+```
+
+#### Fluentd autoscaling:
+
+We have provided an option to enable autoscaling for fluentd deployments. This is disabled by default. 
+
+In order to enable autoscaling for fluentd:
+
+- Enable metrics-server dependency
+  Note: if metrics-server is already installed , this step is not required
+  ```
+  ## Configure metrics-server
+  ## ref: https://github.com/helm/charts/blob/master/stable/metrics-server/values.yaml
+  metrics-server:
+    enabled: true
+  ```
+
+- Enable autoscaling for fluentd
+```
+fluentd:
+    ## Option to turn autoscaling on for fluentd and specify metrics for HPA.
+    autoscaling:
+      enabled: true
+```
+
+
+### Fluentd File-based buffer
+
+By default, we use the in-memory buffer for the Fluentd buffer, but for production environments we recommend users to use the file-based buffer instead.
+
+Buffer configuration can be set in `values.yaml` under the `fluentd` key as follows:
+
+```
+fluentd:
+## Option to specify the Fluentd buffer as file/memory.
+   buffer: 
+     type : "file"
+```
+
+We recommend the following configurations for the file-based buffer:
+
+#### Buffering parameters:
+```
+flush_interval 5s
+
+flush_thread_count 4
+
+chunk_limit_size 100k
+
+total_limit_size 128m
+```
+
+####  Flushing parameters:
+```
+retry_wait 5
+
+retry_forever false
+
+retry_timeout 2h
+
+retry_exponential_backoff_base 2
+```
+
+We have defined several file paths where the buffer chunks will be stored. This parameter must be unique to avoid race condition problem.
+
+Please make sure that you have **enough space in the path directory** . Running out of disk space is a problem frequently reported by users.
+
+Once the config has been modified in the `values.yaml` file,  we need to run the `helm upgrade` in order to apply the changes.
```

Comment:
Change "we" to "you", such as "you need to run"...

Reply:
nit: there's an extra space `... file,  we need`
would also suggest replacing `in order to` with just `to`, either
```you need to run `helm upgrade` to apply the changes.```
or
```you need to run the `helm upgrade` command` to apply the changes.```